src/extensionManager.ts

```diff
@@ -79,15 +82,18 @@ export class ExtensionManager {
     Object.keys(this.extByWorkspace).forEach(key => this.unregisterByName(key))
   }
   shouldStart(workspaceFolderName: string): boolean {
-    const { commonPluginSettings: { enabledWorkspaceFolders, disabledWorkspaceFolders } } = this
+    // tslint:disable-next-line: no-this-assignment
+    const {
+      commonPluginSettings: { enabledWorkspaceFolders, disabledWorkspaceFolders },
+    } = this
     return (
       // start if enabledWorkspaceFolders contains current workspace
       enabledWorkspaceFolders.includes(workspaceFolderName) ||
-      // dont start if disabledWorkspaceFolders doesn't contain current workspace
+      // don't start if disabledWorkspaceFolders doesn't contain current workspace
       !disabledWorkspaceFolders.includes(workspaceFolderName)
     )
   }
-  getByName(workspaceFolderName: string) {
+  public getByName = (workspaceFolderName: string) => {
```

Comment:
arrow-function can capture "this" so it is easier to pass alone as a function-parameter, such as in the `extension.activate()`:
```
context.subscriptions.push(...statusBar.register(extensionManager.getByName),
```
instead of:
```
context.subscriptions.push(...statusBar.register((workspaceFolderName: string) => extensionManager.getByName),
```

Reply:
Ah ok, I didn't notice that.
(Somehow I still slightly prefer your second option or `.register(extensionManager.getByName.bind(extensionManager))` over changing an object method to an error function just that is easier to call it, but that was just my personal stubborn opinion.)